pkg/util/ranger/detacher.go

```diff
@@ -754,8 +754,12 @@ func ExtractEqAndInCondition(sctx *rangerctx.RangerContext, conditions []express
 			}
 			return nil, nil, nil, nil, true
 		} else {
-			// All Intervals are single points
-			accesses[i] = points2EqOrInCond(sctx.ExprCtx, points[i], cols[i])
+			if f, ok := ma.(*expression.ScalarFunction); ok && f.FuncName.L == ast.IsNull {
+				accesses[i] = ma
+			} else {
+				// All Intervals are single points
+				accesses[i] = points2EqOrInCond(sctx.ExprCtx, points[i], cols[i])
+			}
```

Comment:
Do we still need the changes in https://github.com/pingcap/tidb/pull/54253/files#diff-31991a160c8a12cbbd6be603fb144e8028c1f38306bbe2db688427af9fac415bR735?

Reply:
I rewrote this section, and now everything appears to be functioning normally.